apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIdleTasks.java

```diff
@@ -0,0 +1,355 @@
+
+    LOGGER.info(
+        "Starting idle task processing with null mbiHash for: {} Beneficaries and {} Benficiary Histories",
+        beneficiaryCount,
+        historyCount);
+
+    return true;
+  }
+
+  /**
+   * Run this task after the initial task. Respect the TASK_TIME_LIMIT.
+   *
+   * @return true if done with the current task.
+   */
+  public boolean doPostStartupTask() {
+    final Instant startTime = Instant.now();
+    LOGGER.debug("Started a PostStartup time slice");
+
+    // Execute batches in parallel
+    List<Future<Boolean>> tasks = new ArrayList<Future<Boolean>>();
+    tasks.add(executorService.submit(() -> doTransaction(startTime, this::fixupBeneficiaryBatch)));
+    tasks.add(executorService.submit(() -> doTransaction(startTime, this::fixupHistoryBatch)));
+
+    final boolean isDone = waitUntilDone(tasks);
+    LOGGER.debug("Finished a PostStartup time slice");
+    if (isDone) {
+      LOGGER.info("Finished idle startup tasks");
+    }
+    return isDone;
+  }
+
+  /**
+   * Do the normal idle task
+   *
+   * @return true if this task is complete
+   */
+  public boolean doNormalTask() {
+    // Nothing to do normally
+    return true;
+  }
+
+  /**
+   * Wait until all tasks are done or time out
+   *
+   * @param tasks to wait on
+   */
+  private boolean waitUntilDone(List<Future<Boolean>> tasks) {
+    boolean isDone = true;
+    try {
+      // Use a 2x longer value than the expected termination.
+      for (Future<Boolean> task : tasks) {
+        isDone = task.get(2 * TASK_TIME_LIMIT_MILLIS, TimeUnit.MILLISECONDS) && isDone;
+      }
+    } catch (TimeoutException ex) {
+    } catch (ExecutionException ex) {
+    } catch (InterruptedException ex) {
+    }
+    return isDone;
+  }
+
+  /**
+   * Fixup a batch of Beneficiaries. Update the beneficary metrics.
+   *
+   * @param em a {@link EntityManager} setup for a transaction
+   * @return true if done with all fixups
+   */
+  public Boolean fixupBeneficiaryBatch(final EntityManager em, final Instant startTime) {
+    LOGGER.debug("Start fixing up a Beneficiary batch");
+    boolean isDone = true;
+    // Use a cursor, measures slightly faster than
+    try (ScrollableResults itemCursor =
+        em.unwrap(Session.class)
+            .createQuery(SELECT_UNHASHED_BENFICIARIES)
+            .setFetchSize(BATCH_COUNT)
+            .scroll(ScrollMode.SCROLL_INSENSITIVE)) {
+      int count = 0;
+      while (inPeriod(startTime, TASK_TIME_LIMIT) && itemCursor.next()) {
+        final Beneficiary beneficiary = (Beneficiary) itemCursor.get(0);
+        beneficiary
+            .getMedicareBeneficiaryId()
+            .ifPresent(
+                mbi -> {
+                  if (mbi.isEmpty()) return;
+                  final String mbiHash =
+                      "xxx"; // RifLoader.computeMbiHash(options, secretKeyFactory, mbi);
+                  beneficiary.setMbiHash(Optional.of(mbiHash));
+                });
+        isDone = itemCursor.isLast();
```

Comment:
it looks like this could be pulled out of the while loop

Reply:
I don't think so.